### DISCUSSION TITLE
.mmd and .mermaid files not previewing in github.

### DISCUSSION THREAD

**Discussion by valenteEngineering** [community/community#121855] [Programming Help] [answered]
### Select Topic Area

Question

### Body

In my repo we have /documents/mermaid and we have a bunch of .mmd files there. On our local machines we simply open it using IDE plugins or web editor - non-issue.

I would like that within GitHub the 'preview' tab on any of our .mmd files, actually shows the diagram (so our non-technical team can view and edit from source without having to really use git). Viewing a .mmd file within git gives me "Unable to render code block". I have read https://docs.github.com/en/repositories/working-with-files/using-files/working-with-non-code-files#displaying-mermaid-files-on-github but still my .mmd files do not show up.

I understand that in anything markdown supported we can do "```markdown ECT", but that would require us saving as .md files and modifying every one. There must be a way that .mmd files can automatically be previewed but i cant seem to find a  simple solution.  

I may be missing something obvious, but I have yet to find a solution. Any advice would be greatly apricated.

Thank you.

**Comment by MarcosApodaca** [comment, marked as answer]
Github does not support .mmd files, the only way would be to change the format to .md. Markdown 

Here is the list of supported formats
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files

**Reply by ebndev** [reply]
Thanks for taking the time to answer @MarcosApodaca 😁

**Reply by MarcosApodaca** [reply]
Glad to help you 😁

**Reply by ajgrier** [reply]
this makes no sense: if github doesn't support rendering mermaid files, why is it literally reporting a render error? and if mermaid files use markdown, why doesn't github fall back to at least rendering that?

**Reply by cben** [reply]
@MarcosApodaca The doc you linked seems to be about "attaching" a file by drag-and-drop, to have it hosted by github — but not part of git.

Presently, regular .mmd files committed to git do render for me, both
- [x] in repos (https://github.com/cben/sandbox/blob/abfa29d985e5245f067d947d3d75cfbc85c55497/jupyter_notebook/graph.mmd)
- [x] and gist (https://gist.github.com/cben/42255ee5dfda544eb621e2955fc7a99e).

However, I think @valenteEngineering is talking specifically about clicking *Edit* on .mmd files which are part of a repo?  The editor doesn't *Preview* the graph you're editing AFAICT:
- [ ] In a repo, editing [standalone .mmd file](https://github.com/cben/sandbox/blob/master/jupyter_notebook/graph.mmd), the "Preview" is just a source diff, no diagram render.
- [x] In a repo, editing [markdown with \```mermaid fenced blocks](https://github.com/cben/sandbox/blob/06e39cc5f3a34bc2eed1e3fb17cd3801cbee7ad6/README.md#mermaid--mostly-works-in-both-github--gitlab) does Preview the doc with diagram :clap: 
  - [ ] In Preview mode, there is additional "Show diff" checkbox to the right, and that does not diff mermaid blocks, in simply renders the new diagram.  (For comparison, for `$\LaTeX math$` snippets, Show diff renders them twice, old & new formula.)  But that's really nit-picking.
- [ ] In a gist, editing [standalone .mmd file](https://gist.github.com/cben/42255ee5dfda544eb621e2955fc7a99e#file-separate-file-mmd) offers no Preview at all.
- [ ] In a gist, editing [markdown with \```mermaid fenced blocks](https://gist.github.com/cben/42255ee5dfda544eb621e2955fc7a99e#file-mermaid-test-markdown) can Preview the markdown — but leaves the mermaid blocks as literal code blocks (with syntax highlighting :laughing:)
- [x] In repo issue _comments_ (both writing & later Edit), Preview does render \```mermaid fenced blocks.
- [x] Here in discussion comments (both writing & later Edit), Preview does render \```mermaid fenced blocks.
- [ ] In gist _comments_, Preview renders \```mermaid fenced blocks when first submitting the comment — but not when later _Editing_ a comment, it shows as literal code blocks (with syntax highlight)

A workaround is copy-paste to e.g. https://mermaid.live/ and back but in-place preview would be nicer :+1:

**Comment by cben** [comment]
similar or dup: #160296
